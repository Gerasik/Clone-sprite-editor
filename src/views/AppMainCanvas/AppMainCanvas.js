export default class AppMainCanvas {
  constructor(className, parentBlock, size) {
    this.className = className;
    this.parentBlock = parentBlock;
    this.block = null;
    this.size = size;
  }

  render() {
    const mainCanvas = document.createElement('section');
    const canvasImage = document.createElement('canvas');
    const canvasPrev = document.createElement('canvas');
    canvasImage.setAttribute('width', this.size.width);
    canvasImage.setAttribute('height', this.size.width);
    canvasImage.classList.add(`${this.className}__canvas-image`);
    canvasImage.classList.add('canvas-image');
    canvasPrev.setAttribute('width', this.size.width);
    canvasPrev.setAttribute('height', this.size.width);
    canvasPrev.classList.add(`${this.className}__canvas-prev`);
    canvasPrev.classList.add('canvas-prev');
    mainCanvas.classList.add(this.className);
    mainCanvas.appendChild(canvasImage);
    mainCanvas.appendChild(canvasPrev);
    this.parentBlock.appendChild(mainCanvas);
    this.block = mainCanvas;
  }
}
