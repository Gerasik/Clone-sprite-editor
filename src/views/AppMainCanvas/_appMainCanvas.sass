.main-canvas
  flex-grow: 1
  align-self: stretch
  display: flex
  justify-content: center
  align-items: center
  position: relative
  &__canvas-prev,
  &__canvas-image
    image-rendering: pixelated
    max-width: 100%
    min-width: 50%
  &__canvas-prev
    position: absolute
  &__canvas-image
    background-color: white