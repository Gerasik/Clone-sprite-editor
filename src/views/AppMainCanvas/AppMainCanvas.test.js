import AppMainCanvas from './AppMainCanvas';

const context = {
  className: 'main',
  parentBlock: document.body,
  block: null,
  size: 14,
}

describe('new AppMainCanvas', () => {
  const newObj = new AppMainCanvas(context.className, context.parentBlock, context.size);
  it('Should be an instance of Object', () => {
    expect(newObj).toBeInstanceOf(Object);
  });
  it('Should be contain right parameters', () => {
    expect(newObj.className).toBe('main');
    expect(newObj.parentBlock).toBe(document.body);
    expect(newObj.block).toBe(null);
    expect(newObj.size).toBe(14);
  });
});

describe('AppMainCanvas.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppMainCanvas.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render main canvas', () => {
    AppMainCanvas.prototype.render.call(context);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
