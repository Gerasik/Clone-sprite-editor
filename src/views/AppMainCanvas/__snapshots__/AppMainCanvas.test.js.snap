// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMainCanvas.prototype.render Should be render main canvas 1`] = `"<section class=\\"main\\"><canvas width=\\"undefined\\" height=\\"undefined\\" class=\\"main__canvas-image canvas-image\\"></canvas><canvas width=\\"undefined\\" height=\\"undefined\\" class=\\"main__canvas-prev canvas-prev\\"></canvas></section>"`;
